pageserver/src/tenant/timeline/init.rs

```diff
@@ -196,14 +201,21 @@ pub(super) fn cleanup_future_layer(
     name: &LayerFileName,
     disk_consistent_lsn: Lsn,
 ) -> anyhow::Result<()> {
-    use LayerFileName::*;
-    let kind = match name {
-        Delta(_) => "delta",
-        Image(_) => "image",
-    };
     // future image layers are allowed to be produced always for not yet flushed to disk
     // lsns stored in InMemoryLayer.
+    let kind = name.kind();
     tracing::info!("found future {kind} layer {name} disk_consistent_lsn is {disk_consistent_lsn}");
     crate::tenant::timeline::rename_to_backup(path)?;
     Ok(())
 }
+
+pub(super) fn cleanup_local_only_file(
+    path: &Path,
+    name: &LayerFileName,
+    local: &LayerFileMetadata,
+) -> anyhow::Result<()> {
+    let kind = name.kind();
+    tracing::info!("found local-only {kind} layer {name}, metadata {local:?}");
+    crate::tenant::timeline::rename_to_backup(path)?;
```

Comment:
Since this will become relatively common upon pageserver restart/migration, maybe we should delete the file outright instead of renaming it to `.old` -- if one restarted a pageserver a significant number of times, these could take up significant space

Reply:
Thanks, was thinking of the same.

We should probably just do the same with the future layers.

Reply:
As in, I mean that perhaps we should just clear out the backups on restart.

We could of course just delete them straight away. It's hard to see what would we do with these.

Reply:
This is now in commit: [init: actually remove local only and future layers](https://github.com/neondatabase/neon/pull/5198/commits/ac32b8f781d8039b19bf774f7f0087e99743f8f9)

Which will be rebased, but the name will stay.